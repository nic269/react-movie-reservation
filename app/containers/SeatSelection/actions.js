/*
 *
 * SeatSelection actions
 *
 */

import { UPDATE_SEATS_RESERVATION } from './constants';

export function updateSeatsReservation(seats) {
  return {
    type: UPDATE_SEATS_RESERVATION,
    payload: seats,
  };
}
