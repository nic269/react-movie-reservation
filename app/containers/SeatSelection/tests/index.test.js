import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import 'jest-dom/extend-expect';

import configureStore from '../../../configureStore';
import { SeatSelection, mapDispatchToProps } from '../index';
import { updateSeatsReservation } from '../actions';
jest.mock('containers/Room/Loadable', () => 'MockedRoom');

describe('<SeatSelection />', () => {
  let store;
  let props;
  beforeAll(() => {
    props = {
      onSelectSeat: jest.fn(),
      selectedSeats: [
        { id: 'a1', type: 'std' },
        { id: 'c4', type: 'vip' },
        { id: 'd3', type: 'dlx' },
      ],
    };
    store = configureStore({}, browserHistory);
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <SeatSelection {...props} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  it('should call getResult', () => {
    const dispatch = jest.fn();
    const result = mapDispatchToProps(dispatch);
    const seat = { id: 'd3', type: 'dlx' };
    const seats = [
      { id: 'a1', type: 'std' },
      { id: 'c4', type: 'vip' },
      { id: 'd3', type: 'dlx' },
    ];
    result.onSelectSeat(seat, seats);
    expect(dispatch).toHaveBeenCalledWith(updateSeatsReservation(seats));
  });
});
