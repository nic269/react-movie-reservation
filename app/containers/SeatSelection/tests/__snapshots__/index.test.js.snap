// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeatSelection /> Should render and match the snapshot 1`] = `
<div>
  <div>
    <h1>
      Spider-man: Người nhện xa nhà
    </h1>
    <p>
      C13|2D Vietnam sub
    </p>
  </div>
  <mockedroom />
  <div>
    <div>
      <div>
        <h4>
          CGV Crescent Mall
        </h4>
        <p>
          09:10
          ~
          11:10
           | 
          08/07/2019
        </p>
      </div>
      <div>
        <span>
          ₫260,000
        </span>
      </div>
      <button
        type="button"
      >
        Chọn combo
      </button>
      <button
        type="button"
      >
        Thanh toán
      </button>
    </div>
  </div>
</div>
`;
